File: mysql_examples/recipes/api/04_stmt/stmt.py
#!/usr/bin/python
# stmt.py - demonstrate statement processing in Python
# (without placeholders)

import Cookbook
import MySQLdb

try:
  conn = Cookbook.connect ()
except MySQLdb.Error, e:
  print "Cannot connect to server"
  print "Error code:", e.args[0]
  print "Error message:", e.args[1]
  sys.exit (1)

print "Fetch rows with fetchone"
try:
#@ _FETCHONE_
  cursor = conn.cursor ()
  cursor.execute ("SELECT id, name, cats FROM profile")
  while 1:
    row = cursor.fetchone ()
    if row == None:
      break
    print "id: %s, name: %s, cats: %s" % (row[0], row[1], row[2])
  print "Number of rows returned: %d" % cursor.rowcount
  cursor.close ()
#@ _FETCHONE_
except MySQLdb.Error, e:
  print "Oops, the statement failed"
  print e

print "Fetch rows with fetchall"
try:
#@ _FETCHALL_
  cursor = conn.cursor ()
  cursor.execute ("SELECT id, name, cats FROM profile")
  rows = cursor.fetchall ()
  for row in rows:
    print "id: %s, name: %s, cats: %s" % (row[0], row[1], row[2])
  print "Number of rows returned: %d" % cursor.rowcount
  cursor.close ()
#@ _FETCHALL_
except MySQLdb.Error, e:
  print "Oops, the statement failed"
  print e

print "Fetch rows as dictionaries"
try:
#@ _DICT_CURSOR_
  cursor = conn.cursor (MySQLdb.cursors.DictCursor)
  cursor.execute ("SELECT id, name, cats FROM profile")
  for row in cursor.fetchall ():
    print "id: %s, name: %s, cats: %s" % (row["id"], row["name"], row["cats"])
  print "Number of rows returned: %d" % cursor.rowcount
  cursor.close ()
#@ _DICT_CURSOR_
except MySQLdb.Error, e:
  print "Oops, the statement failed"
  print e

print "Execute UPDATE statement (no placeholders)"
try:
#@ _DO_1_
  cursor = conn.cursor ()
  cursor.execute ("UPDATE profile SET cats = cats+1 WHERE name = 'Fred'")
  print "Number of rows updated: %d" % cursor.rowcount
#@ _DO_1_
except MySQLdb.Error, e:
  print "Oops, the statement failed"
  print e

conn.close ()

File: mysql_examples/recipes/api/04_stmt/stmt.php
<?php
# stmt.php - demonstrate statement processing in PHP
# (without placeholders)

require_once "Cookbook.php";

$conn =& Cookbook::connect ();
if (PEAR::isError ($conn))
{
  print ("Cannot connect to server.\n");
  printf ("Error message: %s\n", $conn->getMessage ());
  exit (1);
}

print ("Fetch using ordered array\n");
#@ _FETCHLOOP_1_
$result =& $conn->query ("SELECT id, name, cats FROM profile");
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
while ($row =& $result->fetchRow (DB_FETCHMODE_ORDERED))
  print ("id: $row[0], name: $row[1], cats: $row[2]\n");
printf ("Number of rows returned: %d\n", $result->numRows ());
$result->free ();
#@ _FETCHLOOP_1_

print ("Fetch using associative array\n");
#@ _FETCHLOOP_2_
$result =& $conn->query ("SELECT id, name, cats FROM profile");
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
while ($row =& $result->fetchRow (DB_FETCHMODE_ASSOC))
{
  printf ("id: %s, name: %s, cats: %s\n",
      $row["id"], $row["name"], $row["cats"]);
}
printf ("Number of rows returned: %d\n", $result->numRows ());
$result->free ();
#@ _FETCHLOOP_2_

print ("Fetch using object\n");
#@ _FETCHLOOP_3_
$result =& $conn->query ("SELECT id, name, cats FROM profile");
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
while ($row =& $result->fetchRow (DB_FETCHMODE_OBJECT))
  print ("id: $row->id, name: $row->name, cats: $row->cats\n");
printf ("Number of rows returned: %d\n", $result->numRows ());
$result->free ();
#@ _FETCHLOOP_3_

print ("Fetch using default fetch mode\n");
#@ _SET_FETCH_MODE_
$conn->setFetchMode (DB_FETCHMODE_ASSOC);
#@ _SET_FETCH_MODE_
#@ _FETCHLOOP_4_
$result =& $conn->query ("SELECT id, name, cats FROM profile");
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
while ($row =& $result->fetchRow ())
{
  printf ("id: %s, name: %s, cats: %s\n",
      $row["id"], $row["name"], $row["cats"]);
}
printf ("Number of rows returned: %d\n", $result->numRows ());
$result->free ();
#@ _FETCHLOOP_4_

print ("Update rows\n");
#@ _UPDATE_
$result =& $conn->query ("UPDATE profile SET cats = cats+1
                          WHERE name = 'Fred'");
if (PEAR::isError ($result))
  die ("Oops, the statement failed");
printf ("Number of rows updated: %d\n", $conn->affectedRows ());
#@ _UPDATE_

$conn->disconnect ();
?>
